### PR TITLE
Add separate machine.Start() step to wait for SSH

### PR DESCRIPTION
**Pull request by cfergeau**
This adds a dedicated step in machine.Start() to wait for SSH to become available. And there are also a bunch of minor improvements.

### CODE REVIEW

**Comment by anjannath** [comment]
retest this please

File: pkg/crc/cluster/cluster.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ func WaitForSsh(driver drivers.Driver) error {
 		return nil
 	}
 
-	return errors.RetryAfter(4, checkSshConnectivity, time.Second)
+	return errors.RetryAfter(10, checkSshConnectivity, time.Second)
```
Comment (gbraad) [inline pkg/crc/cluster/cluster.go]: #413

File: pkg/crc/preflight/preflight_checks_common.go [hunk 2]
```diff
@@ -46,3 +49,22 @@ func fixBundleCached() (bool, error) {
 	}
 	return false, fmt.Errorf("CRC bundle is not embedded in the binary, see 'crc help' for more details.")
 }
+
+// Check if oc binary is cached or not
+func checkOcBinaryCached() (bool, error) {
```
Comment (gbraad) [inline pkg/crc/preflight/preflight_checks_common.go]: #414

File: pkg/crc/oc/oc.go [hunk 3]
```diff
@@ -29,7 +29,7 @@ func (oc OcConfig) RunOcCommand(args ...string) (string, string, error) {
 }
 
 // ApproveNodeCSR approves the certificate for the node.
-func ApproveNodeCSR(oc OcConfig) error {
+func (oc OcConfig) ApproveNodeCSR() error {
```
Comment (gbraad) [inline pkg/crc/oc/oc.go]: Please prepend commit message with "Refactor; " else an issue number is needed.

**Pull request merged**